docs/styles/transitions.md

```diff
@@ -20,11 +20,49 @@ and fade out when it is closed:
 }
 ```
 
-In order for the close transition to take effect, you will also need to pass
-the `closeTimeoutMS={150}` prop to each of your modals.
 
 The above example will apply the fade transition globally, affecting all modals
 whose `afterOpen` and `beforeClose` classes have not been set via the
 `className` prop.  To apply the transition to one modal only, you can change
 the above class names and pass an object to your modal's `className` prop as
 described in the [previous section](classes.md).
+
+In order for the close transition to take effect, you will also need to pass
+the `closeTimeoutMS={150}` prop to each of your modals.
+
+Also, if you are using `v3` which supports **React 16**, the close transition works [only if you use](https://github.com/reactjs/react-modal/issues/530#issuecomment-335208533) the `isOpen` prop to toggle the visibility of the modal.
+
+Do not conditionally render the `<Modal />`.
```

Comment:
Maybe we need to elaborate here because it works for react `15`, but not in `16` (if I remember correctly).

It seems that the conditional rendering in react `16+` will destroy the rendered elements unconditionally (seems the reason why it isn't work anymore). So, we probably hold the reference for the modal DOM tree, but it's not attached to any other element (this needs to be confirmed).

Reply:
The example is really awesome. :+1:

Reply:
Hey there

Does this look like a legit explanation to you regarding the limitation to control the unmounting of Portal?
facebook/react#10826

And this leads to the conditional rendering of `<Modal />` not working.

I will try to do some debugging and experiment here.

Reply:
Great! This can be included in the doc.

Reply:
Awesome.

Reply:
You can also answer #530, depending on the conclusion of your investigation and eventually close the issue. I'll continue to follow this and try to help where I can.